docs/source/llm-eval/llm-eval.rst

```diff
@@ -0,0 +1,470 @@
+.. _llm-eval:
```

Comment:
@chenmoneygithub @prithvikannan Can we base this off the https://github.com/mlflow/mlflow/tree/docs-refactor branch and include it in the LLM guide? 

@BenWilson2 can you review the guide from a best practices perspective?

Reply:
This doc can be moved to `docs/source/llms/llm-evaluate/index.rst` and it will serve as the 'landing page' to llm evaluation. I'll leave more comments below about structure.